addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -287,6 +309,18 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
                 last_orderline.set_discount(code.value);
             }
         }
+        /**
+         * Add a product to the current order using the product identifier and lot number from parsed results.
+         * This function retrieves the product identifier and lot number from the `parsed_results` parameter.
+         * It then uses these values to retrieve the product and add it to the current order.
+         */
+        async _barcodeMultipleAction(parsed_results) {
+            const productBarcode = parsed_results.find(element => element.type === 'product');
+            const lotBarcode = parsed_results.find(element => element.type === 'lot');
+            const product = await this._getProductByBarcode(productBarcode);
+            const options = await this._getAddProductOptions(product, lotBarcode);
+            await this.currentOrder.add_product(product, options);
+        }
```

Comment:
How enough is this to only support a gs1 barcode that contains 'product' and 'lot'? Do you need to consider other combinations?

Reply:
After a discussion with the product owner, we found that we can only use product, lot, quantity, and weight in the PoS. But as the quantity and the weight in the barcode represent the lot weight and quantity, they are not useful to add with the barcode.